### DISCUSSION TITLE
Gitlab pages - baseurl & navigation

### DISCUSSION THREAD

**Discussion by freshreplicant** [jpanther/congo#220] [Q&A]
Hi, I'm hosting a hugo site using the congo theme on Gitlab pages and I'm unable to get links to subpages to work properly. The site works perfectly locally, with various links like http://localhost:1313/about/ or http://localhost:1313/blog/ redirecting to the right page. However, on the live Gitlab pages site, click 'About' or 'Blog' just brings me back to the main page. Mousing over I can also see that the link does not redirect to www.mywebsite.com/blog or www.mywebsite.com/about, but to www.mywebsite.com. However, if I manually go to www.mywebsite.com/blog or www.mywebsite.com/about both sites display and work perfectly.

Somebody in the hugo community pointed out to me that this might be due to the baseurl being set to a custom domain, rather than my Gitlab pages URL (see [here](https://gohugo.io/hosting-and-deployment/hosting-on-gitlab/#baseurl)). But changing it to anything but my custom domain totally breaks the theming and the navigation still doesn't work. This leads me to believe that maybe I should leave the baseurl as my custom domain, but I am at a loss how to get the navigation to work properly.

Any advice would be greatly appreciated!

**Comment by jpanther** [comment]
Do you have a repo where I can see the configuration for this site? I have not tried deploying Hugo sites using Gitlab so I'm not entirely sure what the issue might be here.

**Reply by jpanther** [reply]
This issue may also be related to https://github.com/jpanther/congo/issues/230.

**Reply by freshreplicant** [reply]
It's quite possible this is linked in some way. I updated Congo to v2.3.0, tried pushing again but it still seems broken. I then tried changing the baseurl back to "/" rather than my custom domain, but still no luck. Even tried removing the baseurl totally. My website's repo is private sadly, so I can't show the config files.

When I inspect the source of the navigation links, they look like this (bold for emphasis): 

<a class="decoration-primary-500 hover:underline hover:decoration-2 hover:underline-offset-2" **href="" title=""**>About</a>

I'm guessing that's wrong?

Looking through other discussions, it seems maybe #205 was having a similar problem.

EDIT: Ok, some even if I manually navigate to www.mywebsite.com/about, it seems by default the top navigation menu sets every URL to be the same as the page you're on. The bottom navigation with the arrows to the previous page however still work.

**Reply by jpanther** [reply]
What version of Hugo are you using?

**Reply by freshreplicant** [reply]
Locally, I'm running Hugo 0.101.0-1 (Arch Linux), but it all works on localhost. I'm not 100% sure what version Gitlab is on, but I'm fairly sure their docker image is using  0.100.1. 

I'm not sure what else to try at this point, besides maybe removing the theme totally and starting from scratch.

**Reply by jpanther** [reply]
How is your menu configured? It could be that using `pageRef` is not working on your deployment. Do the menu links work if you set the `URL` parameter instead? For instance, using something like `URL = "/about"` instead of `pageRef = "about"`

**Reply by freshreplicant** [reply]
Wow, that actually did it. I have no idea why or how, but navigation works now. The only thing that's not working is the 'Recent' section on the main page, at least the blog entries do show up in their own section. I can do without that for now.

Thanks very much for helping me resolve this issue.

[Edit: Got the Recent section to work, had to edit the mainSections block]
